app/helpers/application_helper.rb

```diff
@@ -55,6 +55,12 @@ def enable_google_analytics(cookies)
       !cookies[CookiesController::NO_ANALYTICS_MODE]
   end
 
+  def using_non_primary_url?
+    return unless ENV["PRIMARY_HOST"].nil?
```

Comment:
ok...(this is painful).. so I think this will 

- return true, if the url you are on is _not_ the PRIMARY_HOST, when that value is defined. 
- return false, if the url you are on _is_  the PRIMARY_HOST, when that value is defined. 

```
  def using_non_primary_url?
    request.host != ENV["PRIMARY_HOST"] if ENV["PRIMARY_HOST"].present?
  end
```

Reply:
So we're _nearly_ there.

`x if y` is the same as:

```
if y
  x
end
```

This is an if statement without an `else` clause, so it doesn't specify what to return if `y` is not true. As such, if `y` is true, that expression will return the value of `x`, but if `y` is not true the expression will return `nil`.

So in this case if PRIMARY_HOST is not set, this method will return `nil`, when ideally we'd like it to return `false`. This is effectively the same thing (because nil is what's called "falsey", so behaves quite a lot _like_ false), but to really tighten this up, I think we're better off ensuring it always returns either `true` or `false`.

I think we're best off using an "and" statement. i.e. we want this method to say "(we know we are using a non-primary URL when) PRIMARY_HOST is present and the request host is different to the PRIMARY_HOST".

To achieve that, I'd recommend:

```ruby
def using_non_primary_url?
   ENV["PRIMARY_HOST"].present? && request.host != ENV["PRIMARY_HOST"]
end
```